framework/physics/physics_material/mgxs/mgxs_simple.cc

```diff
@@ -0,0 +1,70 @@
+// SPDX-FileCopyrightText: 2024 The OpenSn Authors <https://open-sn.github.io/opensn/>
+// SPDX-License-Identifier: MIT
+
+#include "framework/physics/physics_material/mgxs/mgxs.h"
+
+namespace opensn
+{
+
+void
+MGXS::Initialize(unsigned int num_groups, double sigma_t)
+{
+  Reset();
+
+  num_groups_ = num_groups;
+  sigma_t_.resize(num_groups, sigma_t);
+  sigma_a_.resize(num_groups, sigma_t);
+  ComputeDiffusionParameters();
+}
+
+void
+MGXS::Initialize(unsigned int num_groups, double sigma_t, double c)
+{
+  Reset();
+
+  num_groups_ = num_groups;
+  sigma_t_.resize(num_groups, sigma_t);
+  transfer_matrices_.emplace_back(num_groups, num_groups);
+
+  // When multi-group, assign half the scattering cross section
+  // to within-group scattering. The other half will be used for
+  // up/down-scattering.
+  auto& S = transfer_matrices_.back();
+  double scale = (num_groups_ == 1) ? 1.0 : 0.5;
+  S.SetDiagonal(std::vector<double>(num_groups, sigma_t * c * scale));
+
+  // Set the up/down-scattering cross sections.
+  // Summary:
+  //     1) The half of groups with higher energies down-scatter to the next
+  //        lowest energy group half the time and to the same group half the
+  //        time.
+  //     2) The half of groups with lower energies less the last group
+  //        down-scatter to the next lowest energy group three quarters of the
+  //        time and up-scatter to the next highest energy group one quarter
+  //        of the time.
+  //     3) The lowest energy group has the same form as 1).
```

Comment:
why does not the last group upscatter? Strange. Also, how can that group downscatter? It's the last group!!

Reply:
Resolving this in favor of my comment on replacing the entire Initialize(unsigned int num_groups, double sigma_t, double c) routine.